### PR TITLE
nlohmann-json: Update to 3.7.2

### PR DESCRIPTION
**Pull request by Lord-Kamina**
#### Description

@ryandesign asked for an update over at trac, in https://trac.macports.org/ticket/59653

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.6
Xcode 11.1

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?

### CODE REVIEW

**Comment by mf2k** [comment]
You need to add the Trac ticket reference to the commit message so it will get closed automatically. Add this to the end of it:

`Closes: https://trac.macports.org/ticket/59653
`

**Comment by Lord-Kamina** [comment]
> You need to add the Trac ticket reference to the commit message so it will get closed automatically. Add this to the end of it:
> 
> `Closes: https://trac.macports.org/ticket/59653`

Damn. I'm working all day and night, will get it sorted tomorrow.

File: devel/nlohmann-json/Portfile [hunk 1]
```diff
@@ -6,7 +6,7 @@ PortGroup                       github 1.0
 PortGroup                       cmake 1.1
 PortGroup                       cxx11 1.1
```
Comment (reneeotten) [inline devel/nlohmann-json/Portfile]: this PG is obsolete, please remove and add ```compiler.cxx_standard   2011```, probably just before ```depends_build```

File: devel/nlohmann-json/Portfile [hunk 2]
```diff
@@ -16,9 +16,11 @@ maintainers                     {gmail.com:g.litenstein @Lord-Kamina} openmainta
...
-checksums                       rmd160  5cba080556d2383773fe84ab54efb39fafccf1ce \
-                                sha256  f7905c052c00ec615a6e459f2dcc21b7b9c1b2e0f36f4436d0543f15db51ca09 \
-                                size    114794375
+
+checksums                       rmd160  fda813f674ae8aba2a3a1ebd4118f1529bba4fa7 \
+                                sha256  b22dcc77988665cfe9516383cc33d2d1bf64483130240f5475aa553a7e2146d6 \
+                                size    119013338
+
                                 
 depends_build-append            port:gmake
```
Comment (reneeotten) [inline devel/nlohmann-json/Portfile]: you're using the ```cmake``` PG, it seems to me that you don't need to add ```gmake``` here, or...?

**Comment by ra1nb0w** [comment]
@reneeotten wrote me about this PR (see [PR](https://github.com/macports/macports-ports/pull/5775#pullrequestreview-315528647))
I am maintaining the same library `json-modern-cpp` for all Pothos* ports therefore it is useless to have duplicate. I see that the only port that is using `nlohmann-json` is `bali-phy` therefore it is probably easyer to move to `json-modern-cpp` and co-maintain the port (though your port is ~4 months older than mine and therefore you have the precedence). what do you think?

Anyway, you don't need `gmake` and `BUILD_TESTING` (see [line](https://github.com/nlohmann/json/blob/develop/CMakeLists.txt#L82)); moreover use `compiler.cxx_standard 2011` and avoid `cxx11` port group

**Comment by reneeotten** [comment]
@ra1nb0w I don't like the name ```json-modern-cpp``` of "your" port. This library is referred to as ```nlohmann``` by pretty much everyone (as far as I can tell), so I'd prefer to have that in the port name. The other port is only used as a dependency in 5 others, so the amount of work required is not really an argument.

**Comment by ra1nb0w** [comment]
@reneeotten funny, I really think the opposite about the name :)
Ok, seen that everyone else use this name I update Pothos* ports; sorry, I didn't know this fact.

**Comment by Lord-Kamina** [comment]
> Anyway, you don't need `gmake` and `BUILD_TESTING` (see [line](https://github.com/nlohmann/json/blob/develop/CMakeLists.txt#L82)); moreover use `compiler.cxx_standard 2011` and avoid `cxx11` port group

Can't really recall why I added gmake; will try building without it. With regards to testing... I resorted to using an ugly hack because normally the port won't build anything (it's header only) and the tests are entirely optional according to Macports (plus the nlohmann tests take a while to run!), which is why I did it this way.

**Comment by ra1nb0w** [comment]
I did test without `gmake` and `BUILD_TESTING` (the answer is in the code) and it works fine.
`Makefile` is generated at runtime by cmake.

**Comment by Lord-Kamina** [comment]
> I did test without `gmake` and `BUILD_TESTING` (the answer is in the code) and it works fine.
> `Makefile` is generated at runtime by cmake.

Definitely works without gmake. The point of my setup is that it ONLY builds tests when actually intending to run them.

**Pull request merged**